### PR TITLE
More makefile preparations for Windows

### PR DESCRIPTION
**Pull request by Ericson2314**
# Motivation

See each commit for details. In the last one I tried to add some general comments to our entry-point `Makefile` is easier to understand.

# Context

Preparing for #8901. 

Picking up where #9736 left off

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol). 

CC @wegank @Mindavi

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,9 +58,33 @@ else
...
+GLOBAL_CXXFLAGS += -g -Wall -include $(buildprefix)config.h -std=c++2a -I src
+
+# Include the main lib, causing rules to be defined
+
 include mk/lib.mk
 
-# Must be included after `mk/lib.mk` so isn't the default target.
+# Fallback stub rules for better UX when things are disabled
+#
+# These be defined after `mk/lib.mk` because:
```
Comment (Mindavi) [inline Makefile:81]: These must be, or these are

File: Makefile [hunk 2]
```diff
@@ -52,9 +58,33 @@ else
   GLOBAL_CXXFLAGS += -O0 -U_FORTIFY_SOURCE
 endif
 
+include mk/platform.mk
+
+ifdef HOST_WINDOWS
+  # Windows DLLs are stricter about symbol visibilty than Unix shared
+  # objects --- see https://gcc.gnu.org/wiki/Visibility for details.
+  # This is a temporary sledghammer to export everything like on Unix,
```
Comment (Mindavi) [inline Makefile:66]: Yeah, adding exports for all symbols will probably be a bit of work

File: Makefile [hunk 3]
```diff
@@ -52,9 +58,33 @@ else
...
+# Include the main lib, causing rules to be defined
+
 include mk/lib.mk
 
-# Must be included after `mk/lib.mk` so isn't the default target.
+# Fallback stub rules for better UX when things are disabled
+#
+# These be defined after `mk/lib.mk` because:
+#
+# - Otherwise the first one incorrectly becomes the the default target.
```
Comment (Mindavi) [inline Makefile:83]: The the -> the

**Review by Mindavi — APPROVED** [review]
I don't know the reordering implications, but at least have some textual improvements. Otherwise looks good to my eye, but I don't have much experience with these type of makefiles generally.

File: Makefile (hunk 2 above)
Comment (Ericson2314) [reply Makefile]: Same. It's not an urgent todo.

**Pull request merged**